modules/core/src/main/scala/gem/syntax/Parser.scala

```diff
@@ -0,0 +1,24 @@
+// Copyright (c) 2016-2017 Association of Universities for Research in Astronomy, Inc. (AURA)
+// For license information see LICENSE or https://opensource.org/licenses/BSD-3-Clause
+
+package gem.syntax
+
+import atto._, Atto._
+
+final class ParserOps[A](val self: Parser[A]) extends AnyVal {
+
+  /** Parse entire input into an Option. */
+  def parseExact(s: String): Option[A] =
+    (self <~ endOfInput).parseOnly(s).option
```

Comment:
This isn't a combinator really … it's an end of world operation that discharges `Parser` and gives you a normal `Option` back.

But yes, the inner combinator `a <~ endOfInput` does backtrack as expected, and it already exists as `phrase`, so I'll change it! Easy to re-invent things ;-)